crypto/signatures.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2016 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package crypto
+
+import (
+	gocrypto "crypto"
+	"crypto/ecdsa"
+	"crypto/rsa"
+
+	"github.com/google/trillian/crypto/sigpb"
+)
+
+// SignatureAlgorithm returns the algorithm used for this public key.
+// Only ECDSA and RSA keys are supported. Other key types will return sigpb.DigitallySigned_UNKNOWN.
```

Comment:
I think this is intended to mirror [the DigitallySigned struct in RFC5246](https://tools.ietf.org/html/rfc5246#section-7.4.1.4.1), which is why there isn't an "unknown" value. There are cases where anonymous is valid (see that RFC), but never in Trillian. Arguably, this should return an error rather than sigpb.DigitallySigned_ANONYMOUS.